main.go

```
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+type Interval struct {
+	Value float64
+	Unit  string
+}
+
+func getUnits() map[string]string {
+	return map[string]string{
+		"s":  "second",
+		"m":  "minute",
+		"h":  "hour",
+		"d":  "day",
+		"w":  "week",
+		"mo": "month",
+		"y":  "year",
+	}
+}
+
+func ParseInterval(interval string) (*Interval, error) {
```

Add a function `MustParseInterval` as well, that ignores error and returns just the interval